server/handler.go

```diff
@@ -103,6 +106,41 @@ func (h *Handler) ComQuery(
 
 	var r *sqltypes.Result
 	var proccesedAtLeastOneBatch bool
+
+	rowchan := make(chan sql.Row)
+	errchan := make(chan error)
+	quit := make(chan struct{})
+
+	// This goroutine will be select{}ed giving a change to Vitess to call the
+	// handler.CloseConnection callback and enforcing the timeout if configured
+	go func() {
+		for {
+			select {
+			case <-quit:
+				return
+			default:
+			}
+			row, err := rows.Next()
+			if err != nil {
+				errchan <- err
+				return
+			}
+			rowchan <- row
+		}
+	}()
+
+	// Default (big) waitTime is one hour if there is not timeout configured, in which case
+	// it will loop to iterate again. If there is a timeout, it will
+	// be enforced to ensure that Vitess has a chance to call Handler.CloseConnection()
+	waitTime := 1 * time.Hour
+
+	if h.readTimeout > 0 {
+		waitTime = h.readTimeout
+	}
+	timer := time.NewTimer(h.readTimeout)
```

Comment:
this should be waitTime instead of `h.readTimeout`, right?

Reply:
Right, I should avoid doing PRs on fridays, fixes.

Reply:
Right, done.